Preview HTML + bibliographie basée sur Pandoc Export 🎁 🎄

thom4parisot opened the pull request:
Je suis parti de plusieurs principes : 

- prévisualiser et annoter sont 2 actions différentes
- obtenir une prévisualisation avec styles bibliographiques c'est facile grâce à l'export créé par @davidbgk
- il manquait un truc dans l'interface d'écriture pour "faire autre chose" sans ouvrir un panneau de métadonnées

J'ai emprunté le style visuel de #500 (merci @Mogztter et @maiwann 🙏🏻)

Au passage, j'ai corrigé des bugs (certains introduits par #483) : 

- la bibliographie n'était pas sauvegardée
- le store de `workingCopy` n'était pas entièrement initialisé au chargement du composant `Write`
- on ne pouvait pas déplier les détails d'un article avec une navigation clavier
- suppression de re-renders dans le composant `Write`
- correction de l'`ArticleService` où il manquait un câblage entre deux fonctions

Les trucs dont je suis fier et content : 

- si on met à jour la bibliographie (import, suppression de référence), la preview est actualisée automatiquement
- idem si on met à jour une métadonnée

# Todo

- [x] j'ai cassé quelques apparences de bouton en factorisant `.button`, `.primary`, etc. (New Version
- [x] le bouton de fermeture de modale est toujours accessible
- [x] décaler le contenu du panneau latéral de gauche pour que "New Version" n'empiète pas avec le bouton de fermeture
- [x] quand on clique sur une entrée de la table des matières, ça emmène vers le titre dans la preview

## Liste d'articles

![image](https://user-images.githubusercontent.com/138627/209418204-aca151dc-6bdc-4ef7-aa73-5bcc4f6e6470.png)


## Interface d'écriture

![image](https://user-images.githubusercontent.com/138627/209418073-01c60481-6002-483d-9130-90c965051f44.png)

refs #708
refs #676
refs #706

thom4parisot committed:
Toggle article details with keyboard (Space or Enter)

thom4parisot committed:
Article|Book /preview becomes /annotate

thom4parisot committed:
Fix Bibliography service (regression from #483)

thom4parisot committed:
Instate WorkingVersion from #500

thom4parisot committed:
Regroup routes

thom4parisot committed:
Enter proper Preview mode

Also fixes store population which was not very consistent

thom4parisot committed:
Switch between modes, and add other articles actions

thom4parisot committed:
Extract Write GraphQL query from component

davidbgk commented on front/src/hooks/stylo-export.js:
```diff
@@ -28,3 +28,17 @@ export default function useStyloExport (bibStyle) {
 
   return { exportFormats, exportStyles, exportStylesPreview, isLoading }
 }
```
I cannot comment above that line but I think line 27 it should be `bibliography_style: bibStyle` now with the breaking change I introduced in image 0.0.9.

davidbgk commented on front/src/hooks/stylo-export.js:
```diff
@@ -28,3 +28,17 @@ export default function useStyloExport (bibStyle) {
+export function useStyloExportPreview ({ md_content, bib_content, yaml_content }) {
+  const pandocExportEndpoint = useSelector(state => state.applicationConfig.pandocExportEndpoint)
+  const previewArgs = {
+    bibliography_style: 'chicagomodified',
```
That parameter is optional and default to `chicagomodified` on the stylo export view but maybe you prefer to pass it explicitly?

thom4parisot committed:
Fix state value assigned to CSL selector

Without that, after selecting an entry, the first value was always the visually selected one.
We were indeed assigning the "format" value (html, pdf, etc.) which is kind of pointless.

thom4parisot committed:
Use davidbgk/stylo-export new parameter name for CSL

thom4parisot committed:
Add required title for modal, space out, and prevent body scrolling

thom4parisot committed:
Fix relative date formatting

All thanks to DateTime GraphQL scalar which makes it more robust.

see 91dd962ea31b0c3bb170a03bc041ddb25cfe90a3

thom4parisot committed:
Fix updateWorkingVersion nested mutation

I do not pass any resulting field as we do not consume them in the app anyway

thom4parisot committed:
Fix ArticleService output and internal call to GraphQL function

thom4parisot committed:
Ensure props do not trigger re-rendering of children because of unstrict object equality

thom4parisot committed:
Ensure sidebars have the same layout structure

So as they also look similar

thom4parisot committed:
Handle jumping to heading both in Write and Preview
